File: Python/array/fair_candy_swap.py
def fairCandySwap(A: list, B: list) -> list:
    sum_a = sum(A)
    sum_b = sum(B)
    search = set(B)

    result = []
    total = (sum_a + sum_b) // 2
    for a in A:
        if total - sum_a + a in search:
            result = [a, total - sum_a + a]
            break

    return result


if __name__ == "__main__":
    print(fairCandySwap(A = [1,2,5], B = [2,4]))

File: core/tmp/Java/src/com/dbc/algorithm/Array/FairCandySwap.java
package com.dbc.algorithm.Array;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class FairCandySwap {
    public static int[] fairCandySwap(int[] A, int[] B) {
        int sumA = Arrays.stream(A).sum();
        int sumB = Arrays.stream(B).sum();
        int total = (sumA + sumB) / 2;
        Set<Integer> set = new HashSet<>();
        for (int b : B) {
            set.add(b);
        }

        int[] result = new int[2];
        for (int a : A) {
            if (set.contains(total - sumA + a)) {
                result[0] = a;
                result[1] = total - sumA + a;
                break;
            }
        }

        return result;
    }
}
